tests/components/forked_daapd/test_config_flow.py

```diff
@@ -0,0 +1,159 @@
+"""The config flow tests for the forked_daapd media player platform."""
+from asyncio import Future
+
+import pytest
+
+from homeassistant import data_entry_flow
+from homeassistant.components.forked_daapd.const import (
+    CONF_PIPE_CONTROL,
+    CONF_TTS_PAUSE_TIME,
+    CONF_TTS_VOLUME,
+    CONFIG_FLOW_UNIQUE_ID,
+    DOMAIN,
+    FD_NAME,
+)
+from homeassistant.config_entries import (
+    CONN_CLASS_LOCAL_PUSH,
+    SOURCE_USER,
+    SOURCE_ZEROCONF,
+)
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_PORT
+
+from tests.common import MockConfigEntry, patch
+
+SAMPLE_CONFIG = {
+    "websocket_port": 3688,
+    "version": "25.0",
+    "buildoptions": [
+        "ffmpeg",
+        "iTunes XML",
+        "Spotify",
+        "LastFM",
+        "MPD",
+        "Device verification",
+        "Websockets",
+        "ALSA",
+    ],
+}
+
+
+@pytest.fixture(name="config_entry")
+def config_entry_fixture():
+    """Create hass config_entry fixture."""
+    data = {
+        CONF_HOST: "192.168.1.1",
+        CONF_PORT: "2345",
+        CONF_PASSWORD: "",
+    }
+    return MockConfigEntry(
+        version=1,
+        domain=DOMAIN,
+        title="",
+        data=data,
+        options={},
+        system_options={},
+        source=SOURCE_USER,
+        connection_class=CONN_CLASS_LOCAL_PUSH,
+        unique_id=CONFIG_FLOW_UNIQUE_ID,
+        entry_id=1,
+    )
+
+
+async def test_show_form(hass):
+    """Test that the form is served with no input."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == SOURCE_USER
+
+
+async def test_config_flow(hass, config_entry):
+    """Test that the user step works."""
+    with patch(
+        "homeassistant.components.forked_daapd.config_flow.ForkedDaapdAPI.test_connection"
+    ) as mock_test_connection:
+        with patch(
```

Comment:
It's enough with one `with` statement for both patches. Separate the patch instances with comma.